frontend/src/components/GrantRequestPanel/DatabaseResourceForm/common.ts

```diff
@@ -174,3 +178,46 @@ export const flattenTreeOptions = (
     ];
   });
 };
+
+export const parseStringToResource = (
+  key: string
+): DatabaseResource | undefined => {
+  const sections = key.split("/");
+  const resource: DatabaseResource = {
+    databaseName: "",
+  };
+
+  while (sections.length > 0) {
+    const keyword = sections.shift() as DatabaseResourceType | "instances";
+    const data = sections.shift() || "";
+
+    switch (keyword) {
+      case "instances":
+        resource.instanceResourceId = data;
+        break;
+      case "databases":
+        if (!resource.instanceResourceId) {
+          return;
+        }
+        resource.databaseName = `${instanceNamePrefix}${resource.instanceResourceId}/${databaseNamePrefix}${data}`;
```

Comment:
It's good to return the resource identifier from the backend.

On the other hand, in the frontend, it's better to display the human-readable fully qualified table name.

db1.schema1.table1 or db1.table1

The current display is a regression from the previous one. especially when schema is empty as there are double `//` in `schemas//tables/foo`

Reply:
I prefer to use `databases/{db}/tables/{table}` format cause this will unify with the frontend. Create a PR to optimize the error message https://github.com/bytebase/bytebase/pull/14237